src/main/java/com/google/devtools/build/lib/rules/android/AndroidRuleClasses.java

```diff
@@ -374,7 +374,6 @@ public static boolean hasProguardSpecs(AttributeMap rule) {
 
           if (AndroidResources.definesAndroidResources(attributes)) {
             implicitOutputs.add(
-                AndroidRuleClasses.ANDROID_JAVA_SOURCE_JAR,
```

Comment:
Removing this will break any current usage of the .srcjar output as a label, whether the flag is on or off, so it looks like we'd need to go through the incompatible change process. Unfortunately there's no way to conditionally register implicit outputs based on configuration. Glancing through the code, it should be technically possible to enable that, but we'd need to think through that separately. The next best thing is to leave this here, and when --link_library_resources is false, register a FailAction for that output. This means that the failure won't be surfaced until execution time, unfortunately.